### PR TITLE
Bulk generate playlists for all users using Spark

### PR DESCRIPTION
**Pull request by amCap1712**
Troi is good for experimenting with playlist patches but is too slow for generating those for all users. Most of the data troi needs to generate its playlists is already available in spark, making it simple to do the bulk generation there.

### CODE REVIEW

File: listenbrainz_spark/troi/periodic_jams.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    elif slug == "weekly-exploration":
+        time_filter = "latest_listened_at IS NULL"
+    else:
+        return []
+
+    query = f"""
+        WITH recommendations AS (
+            SELECT user_id
+                 , recording_mbid
+                 , rank() OVER (PARTITION BY user_id ORDER BY score DESC) AS position
```
Comment (mayhem) [inline listenbrainz_spark/troi/periodic_jams.py:28]: the recommendation tracks should be sorted randomly, not by order. In troi the playlist gets shuffled before being truncated, but that is hidden in the playlist generator.

**Review by mayhem — CHANGES_REQUESTED** [review]
One minor sort change and that should be it!

File: listenbrainz_spark/troi/periodic_jams.py (hunk 1 above)
Comment (amCap1712) [reply listenbrainz_spark/troi/periodic_jams.py:28]: Shuffle before truncating in all cases?

Comment (mayhem) [reply listenbrainz_spark/troi/periodic_jams.py:28]: I actually went back and looked at exactly what troi does and its the other way around. Sort by ranking, truncate THEN shuffle. Not how I had originally intended, but it works well, so let's continue to do that.

Comment (amCap1712) [reply listenbrainz_spark/troi/periodic_jams.py]: Fixed.

**Review by mayhem — APPROVED** [review]
OK, lets get this merged. We'll still need a crontab entry, yes? I think we should run this daily for a while until it is clear that the process is stable. Then we drop back to weekly. How does that sound?

**Pull request merged**